### PR TITLE
Update message handling to use insertion sort

### PR DESCRIPTION
**Pull request by huffman**

### CODE REVIEW

**Comment by huffman** [comment]
I NEED TO TEST THIS, DON'T MERGE

**Comment by birarda** [comment]
Okay, merging

File: libraries/networking/src/udt/Connection.cpp [hunk 1]
```diff
@@ -825,30 +825,35 @@ void PendingReceivedMessage::enqueuePacket(std::unique_ptr<Packet> packet) {
...
-    _packets.push_back(std::move(packet));
+    // Insert into the packets list in sorted order. Because we generally expect to receive packets in order, begin
+    // searching from the end of the list.
+    auto it = _packets.rbegin();
+    for (auto rend = _packets.rend(); it != rend; ++it) {
+        if (sequenceNumber > (*it)->getSequenceNumber()) {
+            break;
+        }
+    }
+    _packets.insert(it.base(), std::move(packet));
```
Comment (Atlante45) [inline libraries/networking/src/udt/Connection.cpp:848]: Nice, I didn't know the `base` method. Can be useful.
You could probably use `std::find` above.

Comment (huffman) [inline libraries/networking/src/udt/Connection.cpp:848]: Good idea, I'll try updating this to use std::find. When I put this in I thought "raw loop, hmm..." but wasn't sure how to replace it.

**Pull request merged**